### PR TITLE
Fix handling none source in build_torch_function_fn 

### PR DESCRIPTION
**Pull request by laithsakka**
Fix https://github.com/pytorch/pytorch/issues/119580

When a UserDefinedObjectVariable is created it does not always have a source, i.e: when its an intermediate
This diff fix two handling of none source in two locations during an inlining of a user torch function.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119724

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by jansel — APPROVED** [review]

**Review by mlazos — APPROVED** [review]

**Comment by laithsakka** [comment]
ah just saw the failures looking at them

**Comment by laithsakka** [comment]
ah there are Unexpected success, easy

**Review by anijain2305 — APPROVED** [review]

File: torch/_dynamo/variables/torch_function.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def call_torch_function(
 def build_torch_function_fn(tx, value, source):
     from .builder import SourcelessBuilder, VariableBuilder
 
-    if not source:
+    if source:
```
Comment (anijain2305) [inline torch/_dynamo/variables/torch_function.py:100]: Haha!

**Comment by laithsakka** [comment]
rebase

**Pull request closed without merging**